cmake/OpenCVDownload.cmake

```diff
@@ -37,6 +37,47 @@ file(WRITE "${OPENCV_DOWNLOAD_LOG}" "#use_cache \"${OPENCV_DOWNLOAD_PATH}\"\n")
 file(REMOVE "${OPENCV_DOWNLOAD_WITH_CURL}")
 file(REMOVE "${OPENCV_DOWNLOAD_WITH_WGET}")
 
+# host
+set(OPENCV_DOWNLOAD_HOST "GITHUB" CACHE INTERNAL "Used for 3rdparty archives")
+# mirrors
+set(OPENCV_MIRROR_GITCODE "gitcode.net" CACHE INTERNAL "Link to mirror hosted by gitcode")
+set(OPENCV_MIRROR_CUSTOM "" CACHE INTERNAL "Link to another mirror hosted in gitlab-style")
+
+function(ocv_init_download)
+  # Run `git remote get-url origin` to get remote source
+  execute_process(
+    COMMAND
+      git remote get-url origin
+    RESULT_VARIABLE
+      RESULT_STATUS
+    OUTPUT_VARIABLE
+      OCV_GIT_ORIGIN_URL_OUT
+    ERROR_QUIET
+  )
+  # if non-git, OCV_GIT_ORIGIN_URL_OUT is empty
+  if(NOT OCV_GIT_ORIGIN_URL_OUT)
+    message(STATUS "ocv_init_download: This is not a git repo. 3rdparty resources will be downloaded from Github.com by default. You can use mirrors with CMake option -DOPENCV_MIRROR_CUSTOM=gitcode.net for example.")
```

Comment:
> This is not a git repo

"OpenCV source tree is not fetched as git repository"?
"Unable to detect git repository server of OpenCV source tree"?

Reply:
Done. Replaced with "OpenCV source tree is not fetched as git repository".